clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/lifecycle/LaunchFailureNotificationCleanupAgent.java

```diff
@@ -0,0 +1,179 @@
+import com.netflix.spinnaker.cats.agent.RunnableAgent;
+import com.netflix.spinnaker.clouddriver.aws.AmazonCloudProvider;
+import com.netflix.spinnaker.clouddriver.aws.provider.AwsProvider;
+import com.netflix.spinnaker.clouddriver.aws.security.AmazonClientProvider;
+import com.netflix.spinnaker.clouddriver.aws.security.NetflixAmazonCredentials;
+import com.netflix.spinnaker.clouddriver.cache.CustomScheduledAgent;
+import com.netflix.spinnaker.clouddriver.model.EntityTags;
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsProvider;
+import com.netflix.spinnaker.clouddriver.tags.ServerGroupTagger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.UndeclaredThrowableException;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class LaunchFailureNotificationCleanupAgent implements RunnableAgent, CustomScheduledAgent {
+  private static final Logger log = LoggerFactory.getLogger(LaunchFailureNotificationAgent.class);
+
+  private static final String TAG_NAME = "spinnaker_ui_alert:autoscaling:ec2_instance_launch_error";
+  private static final int MAX_RESULTS = 10000;
+
+  private final AmazonClientProvider amazonClientProvider;
+  private final AccountCredentialsProvider accountCredentialsProvider;
+  private final ServerGroupTagger serverGroupTagger;
+
+  LaunchFailureNotificationCleanupAgent(AmazonClientProvider amazonClientProvider,
+                                        AccountCredentialsProvider accountCredentialsProvider,
+                                        ServerGroupTagger serverGroupTagger) {
+    this.amazonClientProvider = amazonClientProvider;
+    this.accountCredentialsProvider = accountCredentialsProvider;
+    this.serverGroupTagger = serverGroupTagger;
+  }
+
+  @Override
+  public String getAgentType() {
+    return LaunchFailureNotificationCleanupAgent.class.getSimpleName();
+  }
+
+  @Override
+  public String getProviderName() {
+    return AwsProvider.PROVIDER_NAME;
+  }
+
+  @Override
+  public long getPollIntervalMillis() {
+    return TimeUnit.MINUTES.toMillis(5);
+  }
+
+  @Override
+  public long getTimeoutMillis() {
+    return -1;
+  }
+
+  @Override
+  public void run() {
+    Collection<EntityTags> taggedEntities = serverGroupTagger.taggedEntities(
+      AmazonCloudProvider.ID,
+      null, // all accounts
+      TAG_NAME,
+      MAX_RESULTS
+    );
+
+    Map<String, AmazonAutoScaling> amazonAutoScalingClients = new HashMap<>();
+
+    taggedEntities.forEach(entityTags -> {
+      EntityTags.EntityRef entityRef = entityTags.getEntityRef();
+
+      AmazonAutoScaling amazonAutoScaling = getAmazonAutoScalingClient(
+        amazonAutoScalingClients, entityRef.getAccount(), entityRef.getRegion()
+      );
+
+      try {
+        if (hasLaunchFailures(amazonAutoScaling, entityTags)) {
```

Comment:
@cfieber Wasn't sure if it was worth an additional `RateLimiter` here if there's fear that fetching scaling activities will blow a limit.

I don't really expect there to ever be _many_ server groups in this state.

(_for example_) after a couple weeks of being live there were ~30-40 server groups that got cleaned up.